src/SystemReport.php

```diff
@@ -0,0 +1,248 @@
+	private $excluded = array();
+
+	/**
+	 * SystemReport constructor.
+	 *
+	 * @param string $id The gateway ID.
+	 * @param string $name The gateway name (or title).
+	 */
+	public function __construct( $id, $name ) {
+		$this->id   = $id;
+		$this->name = $name;
+		$this->register( $id );
+
+		add_action( 'woocommerce_system_status_report', array( $this, 'add_status_page_box' ) );
+		add_action( 'woocommerce_cleanup_logs', array( __CLASS__, 'remove_old_entries' ) );
+	}
+
+	/**
+	 * Register gateway gateway system report.
+	 *
+	 * @param string $id The gateway ID.
+	 */
+	private function register( $id ) {
+		$registry = get_option( self::REGISTRY_OPTION, array() );
+		if ( isset( $registry[ $id ] ) ) {
+			return;
+		}
+
+		$registry[] = $id;
+		update_option( self::REGISTRY_OPTION, $registry );
+	}
+
+	/**
+	 * Get the current settings for the gateway.
+	 *
+	 * @return array|false An associative title:value array of the current settings. False if plugin settings is cannot be retrieved.
+	 */
+	private function get_current_settings() {
+		$tracked = array(
+			'text',
+			'textarea',
+			'select',
+			'multiselect',
+			'radio',
+			'checkbox',
+			'number',
+			'email',
+			'tel',
+			'url',
+			'color',
+			'date',
+			'file',
+		);
+
+		$skipped = array_merge( array( 'title' => 'enable/disable' ), $this->excluded );
+
+		$payment_gateways = WC()->payment_gateways()->payment_gateways();
+		$gateway          = $payment_gateways[ $this->id ] ?? null;
+		$form_fields      = $gateway ? $gateway->get_form_fields() : array();
+
+		if ( empty( $form_fields ) ) {
+			return false;
+		}
+
+		$output   = array();
+		$settings = get_option( 'woocommerce_' . $this->id . '_settings', array() );
+		foreach ( $settings as $setting_key => $value ) {
+			$form_field = $form_fields[ $setting_key ];
+
+			if ( ! empty( $this->included ) ) {
+				foreach ( $this->included as $setting ) {
+					if ( is_array( $setting ) ) {
+						if ( $setting['value'] !== $form_field[ $setting['key'] ] ) {
+							continue;
+						}
+					} elseif ( $setting !== $setting_key ) {
+							continue;
+					}
+				}
+			} else {
+
+				if ( ! in_array( $form_field['type'], $tracked, true ) ) {
+					continue;
+				}
+
+				if ( ! isset( $form_field['title'] ) ) {
+					continue;
+				}
+
+				if ( in_array( strtolower( $form_field['title'] ), $skipped, true ) ) {
+					continue;
+				}
+
+				foreach ( $skipped as $setting ) {
+					// Skip based on specific form field key, and value.
+					if ( is_array( $setting ) ) {
+						if ( $setting['value'] === $form_field[ $setting['key'] ] ) {
+							continue;
+						}
+						// Skip based on setting name.
+					} elseif ( $skipped === $setting_key ) {
+						continue;
+					}
+				}
+			}
+
+			if ( empty( $value ) ) {
+				$value = $form_field['default'] ?? $value;
+			}
+
+			$output[ $setting_key ] = array(
+				'title' => rtrim( $form_field['title'], ':' ),
+				'value' => $value,
+			);
+
+		}
+
+		return $output;
+	}
```

Comment:
This is obviously very limited to just gateways here. Right now this would only work for gateways and no other implementation of the settings API from WooCommerce. Maybe we can look at refactoring this now or in the future to make it able to support more type of integrations?

Reply:
I suggest we'll take a look into this in the future as it is not yet used in any plugin.